### PR TITLE
feat: new device classes for new sensor types

### PR DESCRIPTION
**Pull request by Ernst79**
new device classes for new sensor types

### CODE REVIEW

File: src/sensor_state_data/library.py [hunk 1]
```diff
@@ -71,22 +71,22 @@ class SensorLibrary:
         )
     )
     WEIGHT__MASS_KILOGRAMS = BaseSensorDescription(
-        device_class=None,
+        device_class=SensorDeviceClass.WEIGHT,
```
Comment (bdraco) [inline src/sensor_state_data/library.py:74]: Should this be MASS or WEIGHT?

File: src/sensor_state_data/sensor/device_class.py [hunk 2]
```diff
@@ -98,3 +107,6 @@ class SensorDeviceClass(BaseDeviceClass):
 
     # voltage (V)
     VOLTAGE = "voltage"
+
+    # weight (g, kg, lbs)
+    WEIGHT = "weight"
```
Comment (bdraco) [inline src/sensor_state_data/sensor/device_class.py:112]: Should we call this MASS ?

Comment (Ernst79) [reply src/sensor_state_data/sensor/device_class.py]: yes, you're right. Changed it.

File: src/sensor_state_data/library.py (hunk 1 above)
Comment (Ernst79) [reply src/sensor_state_data/library.py]: Yes, we do. Changed it.

File: src/sensor_state_data/library.py [hunk 3]
```diff
@@ -70,23 +70,23 @@ class SensorLibrary:
             native_unit_of_measurement=Units.CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         )
     )
-    WEIGHT__MASS_KILOGRAMS = BaseSensorDescription(
-        device_class=None,
+    MASS_WEIGHT__MASS_KILOGRAMS = BaseSensorDescription(
+        device_class=SensorDeviceClass.MASS_WEIGHT,
```
Comment (bdraco) [inline src/sensor_state_data/library.py:74]:
```suggestion
        device_class=SensorDeviceClass.MASS,
```

File: src/sensor_state_data/library.py [hunk 4]
```diff
@@ -70,23 +70,23 @@ class SensorLibrary:
             native_unit_of_measurement=Units.CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         )
     )
-    WEIGHT__MASS_KILOGRAMS = BaseSensorDescription(
-        device_class=None,
+    MASS_WEIGHT__MASS_KILOGRAMS = BaseSensorDescription(
```
Comment (bdraco) [inline src/sensor_state_data/library.py:73]:
```suggestion
    MASS__MASS_KILOGRAMS = BaseSensorDescription(
```

File: src/sensor_state_data/library.py [hunk 5]
```diff
@@ -70,23 +70,23 @@ class SensorLibrary:
...
+        device_class=SensorDeviceClass.MASS_WEIGHT,
         native_unit_of_measurement=Units.MASS_KILOGRAMS,
     )
-    WEIGHT__MASS_POUNDS = BaseSensorDescription(
-        device_class=None,
+    MASS_WEIGHT__MASS_POUNDS = BaseSensorDescription(
```
Comment (bdraco) [inline src/sensor_state_data/library.py:77]:
```suggestion
    MASS__MASS_POUNDS = BaseSensorDescription(
```

File: src/sensor_state_data/library.py [hunk 6]
```diff
@@ -70,23 +70,23 @@ class SensorLibrary:
...
+        device_class=SensorDeviceClass.MASS_WEIGHT,
```
Comment (bdraco) [inline src/sensor_state_data/library.py:78]:
```suggestion
        device_class=SensorDeviceClass.MASS,
```

File: src/sensor_state_data/sensor/device_class.py [hunk 7]
```diff
@@ -45,6 +51,12 @@ class SensorDeviceClass(BaseDeviceClass):
     # current light level (lx/lm)
     ILLUMINANCE = "illuminance"
 
+    # % of moisture in the air or soil
+    MOISTURE = "moisture"
+
+    # mass weight (g, kg, lbs)
+    MASS_WEIGHT = "mass_weight"
```
Comment (bdraco) [inline src/sensor_state_data/sensor/device_class.py:58]:
```suggestion
    MASS = "mass"
```

**Comment by bdraco** [comment]
I think we are talking about measuring mass here and not weight.

mass and weight are always so confusing though 

https://www.snexplores.org/article/explainer-how-do-mass-and-weight-differ#:~:text=Mass%20is%20a%20fundamental%20measurement%20of%20how%20much%20matter%20an%20object%20contains.&text=Weight%20is%20a%20measurement%20of,actually%20a%20measure%20of%20force.

**Comment by bdraco** [comment]
Trying not to be pedantic here but since this is so hard to change later, I think we need to be careful to get it right the first go.

**Comment by Ernst79** [comment]
Better do it right now, I agree. It's just that I never use mass, when I step on a scale 😄

**Review by bdraco — APPROVED** [review]

**Pull request merged**